shell/layouts/default.vue

```diff
@@ -194,224 +209,101 @@ export default {
+      const currentProduct = this.productId; // Use the productId computed property as a dependency to trigger updates when the product changes.
 
-          await this.$store.dispatch('prefs/setLastVisited', route);
+      if (this.isExplorer) {
+        for (const product of this.activeProducts) {
+          if (product.inStore === 'cluster') {
+            addObject(loadProducts, product.name);
+          }
         }
       }
-    },
-
-    $route(a, b) {
-      this.$nextTick(() => this.syncNav());
-    },
-
-  },
-
-  async created() {
-    this.queueUpdate = debounce(this.getGroups, 500);
-
-    this.getGroups();
-
-    await this.$store.dispatch('prefs/setLastVisited', this.$route);
-  },
-
-  mounted() {
-    // Sync the navigation tree on fresh load
-    this.$nextTick(() => this.syncNav());
-  },
-
-  methods: {
-    async setClusterAsLastRoute() {
-      const route = {
-        name:   this.$route.name,
-        params: {
-          ...this.$route.params,
-          cluster: this.clusterId,
-        }
-      };
 
-      await this.$store.dispatch('prefs/setLastVisited', route);
-    },
-    handlePageAction(action) {
-      if (action.action === SET_LOGIN_ACTION) {
-        this.afterLoginRoute = this.getLoginRoute();
-        // Mark release notes as seen, so that the login route is honoured
-        markSeenReleaseNotes(this.$store);
-      }
-    },
+      // This should already have come into the list from above, but in case it hasn't...
+      addObject(loadProducts, currentProduct);
 
-    getLoginRoute() {
-      return {
-        name:   this.$route.name,
-        params: this.$route.params
-      };
+      return loadProducts;
     },
 
-    collapseAll() {
-      this.$refs.groups.forEach((grp) => {
-        grp.isExpanded = false;
-      });
+    currentType() {
+      return this.$route.params.resource || '';
     },
 
-    getGroups() {
-      if ( this.gettingGroups ) {
-        return;
-      }
-      this.gettingGroups = true;
-
-      if ( !this.clusterReady ) {
-        clear(this.groups);
-        this.gettingGroups = false;
-
-        return;
+    groups() {
+      if (!this.clusterReady || !this.allSchemas || !this.counts) {
+        // Referencing these computed properties
+        // to trigger the nav tree to be recomputed if they change.
+        return [];
       }
-
-      const clusterId = this.$store.getters['clusterId'];
-      const currentProduct = this.$store.getters['productId'];
-      const currentType = this.$route.params.resource || '';
       let namespaces = null;
 
+      // Always show cluster-level types, regardless of the namespace filter
       if ( !this.$store.getters['isAllNamespaces'] ) {
         const namespacesObject = this.$store.getters['namespaces']();
 
         namespaces = Object.keys(namespacesObject);
       }
-
-      // Always show cluster-level types, regardless of the namespace filter
-      const namespaceMode = 'both';
```

Comment:
Removed the `namespaceMode` here because there was already a computed property for it which was 'both' by default.